package NKikimrColumnShardDataSharingProto;

message TInitiator {
    message TController {
        optional string ClassName = 1;

        message TTest {
        }

        message TSchemeShard {
            optional uint64 TabletId = 1;
        }

        oneof Implementation {
            TTest Test = 40;
            TSchemeShard SS = 41;
        }
    }

    message TStatus {
        message TInProgress {
            optional uint32 PortionsCount = 1;
            optional uint32 PortionsReady = 2;
        }

        message TStartFailed {
            optional string ErrorMessage = 1;
        }

        message TNotFound {
            optional string ErrorMessage = 1;
        }

        optional string ClassName = 1;
        optional string SessionId = 2;

        oneof Implementation {
            TInProgress InProgress = 40;
            TStartFailed StartFailed = 41;
            TNotFound NotFound = 42;
        }
    }
}
